modules/parsers/metadata.js

```diff
@@ -1,13 +1,16 @@
+import blockParser from './block.js';
+
 /* global WebImporter */
 
 function isDate(str) {
+  if (typeof str !== 'string') return false;
   const date = new Date(str);
-  return !Number.isNaN(date);
+  return !Number.isNaN(Number(date));
 }
 
-export default function parse(element, { document, params: { metadata = {} } }) {
+export default function parse(element, { document, params: { cells = {} } }) {
   const baseMetadata = WebImporter.Blocks.getMetadata(document) || {};
-  const customMetadata = WebImporter.Transformer.buildBlockConfig(document, metadata);
+  const customMetadata = blockParser(document, { params: { cells } });
```

Comment:
Treat `metadata` block like any other block which means the generic block parser can be used.